tests/custom_strategies/__init__.py

```diff
@@ -358,3 +360,158 @@ def valid_shapes(
         if len(shape) == 1 and draw(st.booleans())
         else tuple(int(i) for i in shape)
     )
+
+
+@st.composite
+def arbitrary_indices(draw, shape: Tuple[int]):
+    """
+    Hypothesis search strategy: Generate a valid index
+    for an array of a given shape. The index can contain
+    any type of valid object used for indexing, including
+    integers, slices, Ellipsis's, newaxis's, boolean arrays,
+    and integer arrays.
+
+    Parameters
+    ----------
+    shape : Tuple[int]
+        The shape of the array to be indexed into
+
+    Notes
+    -----
+    `draw` is a parameter reserved by hypothesis, and should not be specified
+    by the user.
+
+    Returns
+    -------
+    hypothesis.searchstrategy.SearchStrategy[Tuple[Union[int, slice, Ellipsis, NoneType, numpy.ndarray], ...]]
+    """
+
+    def group_continuous_integers(ls):
+        """
+        Given a list of integers, find and group continuous sequences
+
+        Parameters
+        ----------
+        ls: List[int]
+
+        Returns
+        -------
+        List[Tuple[int]]
+
+        Examples
+        --------
+        >>> group_continuous_integers([1, 3, 4, 5, 7, 8])
+        [(1,), (3, 4, 5), (7, 8)]
+        """
+        return [
+            tuple(map(itemgetter(1), g))
+            for k, g in groupby(enumerate(ls), lambda x: x[0] - x[1])
+        ]
+
```

Comment:
Ahh nice catch. In the case of empty arrays, I think this fix will make it so that the case

```
np.array([])[[]]
```

could never be checked (though for what reason this would ever need to be used, I don't know). Since the `integer_array_indices` strategy doesn't handle dim-0 axes, I can add a small test to `test_getitem` to check for this